File: backend/components/metiers/threed/utils/plugins/core/pluginManager.js
// pluginManager.js – Gestionnaire de plugins avancé pour Threed (threed)
// Fichier déplacé dans samples/pluginManager.js
class PluginManager {
  constructor() {
    this.plugins = [];
  }
  register(plugin) {
    this.plugins.push(plugin);
  }
  runAll(...args) {
    return this.plugins.map(plugin => plugin(...args));
  }
  list() {
    return this.plugins.map(p => p.name || 'anonymous');
  }
}

const pluginManager = new PluginManager();
module.exports = { pluginManager };

File: backend/components/metiers/threed/utils/plugins/core/pluginManager.py
"""
pluginManager.py – Gestionnaire de plugins pour Threed (Python)
"""


class PluginManager:
    def __init__(self):
        self.plugins = []

    def register(self, plugin):
        self.plugins.append(plugin)

    def run_all(self, *args, **kwargs):
        results = []
        for plugin in self.plugins:
            results.append(plugin(*args, **kwargs))
        return results


pluginManager = PluginManager()

# Fichier déplacé dans samples/pluginManager.py
